app/src/main/java/org/sagebionetworks/research/mpower/studyburst/StudyBurstActivity.kt

```diff
@@ -0,0 +1,96 @@
+package org.sagebionetworks.research.mpower.studyburst
+
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.activity_study_burst.*
+import kotlinx.android.synthetic.main.study_burst_cell.view.*
+import org.slf4j.LoggerFactory
+
+import org.sagebionetworks.research.mpower.R
+import org.sagebionetworks.research.mpower.TaskLauncher
+import javax.inject.Inject
+
+
+class StudyBurstActivity : AppCompatActivity() {
+    private val LOGGER = LoggerFactory.getLogger(StudyBurstActivity::class.java)
+
+    private val studyBurstViewModel: StudyBurstViewModel by lazy {
+        ViewModelProviders.of(this).get(StudyBurstViewModel::class.java)
+    }
+
+    @Inject
+    lateinit var taskLauncher: TaskLauncher
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+        LOGGER.debug("StudyBurstActivity.onCreate()")
+        setContentView(R.layout.activity_study_burst)
+
+        studyBurstViewModel.getMessage().observe(this, Observer {
+            text -> studyBurstMessage.setText(text)
+        })
+        studyBurstViewModel.getTitle().observe(this, Observer {
+            text -> studyBurstTitle.setText(text)
+        })
+        studyBurstViewModel.getExpires().observe(this, Observer {
+            text -> expiresText.setText(getResources().getString(R.string.study_burst_progress_message, text))
+        })
+
+        studyBurstViewModel.getItems().observe(this, Observer {
+            items ->
+                LOGGER.debug("How many items: " + items?.size)
+                var completed = items?.count { it.completed }
```

Comment:
var completed = items?.count { it.completed } ?: arrayOf()

Reply:
oops wrong return type, use this...

var completed = items?.count { it.completed } ?: 0